src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -770,6 +777,8 @@ private bool ExecuteClusterTransactionOneByOne(ArraySegment<ClusterTransactionCo
 
                     DatabaseShutdown.WaitHandle.WaitOne(_clusterTransactionDelayOnFailure);
                     _clusterTransactionDelayOnFailure = Math.Min(_clusterTransactionDelayOnFailure * 2, 15000);
+                    _hasClusterTransaction.Set();
```

Comment:
When we return 0 as the number of commands that were executed we assume no more commands left.
This is an edge case, but I set it here to avoid a scenario in which we don't execute the last command.
This will happen only if no other commands are added to the queue (and they will trigger `_hasClusterTransaction.Set()` in `NotifyOnPendingClusterTransaction`).